### PR TITLE
homer: init 24.11.4

### LINKED ISSUE #152343: Package Homer
**Project description**
> A dead simple static HOMepage for your servER to keep your services on hand, from a simple yaml configuration file.
>
> 

Homer is a homepage to link to other hosted services.
As the configuration is all done in YAML, it shouldn't be too hard to represent in a Nix derivation.

**Metadata**

* homepage URL: https://github.com/bastienwirtz/homer
* source URL: https://github.com/bastienwirtz/homer
* license: Apache-2.0
* platforms: linux, maybe unix & darwin

### PR DESCRIPTION
**Pull request by Stunkymonkey**
fix #152343

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
ping @tecosaur how did you come up with the patch? it does not build for me.

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 354293` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>homer</li>
  </ul>
</details>

File: pkgs/by-name/ho/homer/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  fetchFromGitHub,
+  pnpm,
+  nodejs,
+  pkgs,
+}:
+pkgs.stdenv.mkDerivation rec {
```
Comment (PatrickDaG) [inline pkgs/by-name/ho/homer/package.nix:8]: I think you could switch to `stdenvNoCC` as you don't need any c compilers.

File: pkgs/by-name/ho/homer/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  fetchFromGitHub,
+  pnpm,
+  nodejs,
+  pkgs,
```
Comment (PatrickDaG) [inline pkgs/by-name/ho/homer/package.nix:6]: You shouldn't do the include the whole `pkgs` here but only the ones you need.

File: pkgs/by-name/ho/homer/package.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      version
+      src
+      patches
+      ;
+    hash = "sha256-5unoY8lPaX9sZAJEBICpxSddwLV8liK1tbamB2ulvew=";
+  };
+
+  patches = [ ./sass-embedded.patch ];
+
+  nativeBuildInputs = with pkgs; [
```
Comment (PatrickDaG) [inline pkgs/by-name/ho/homer/package.nix:30]: `with pkgs` shouldn't be used as it's unintuitive in regards to which package is in scope.

File: pkgs/by-name/ho/homer/package.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    inherit
+      pname
+      version
+      src
+      patches
+      ;
+    hash = "sha256-5unoY8lPaX9sZAJEBICpxSddwLV8liK1tbamB2ulvew=";
+  };
+
+  patches = [ ./sass-embedded.patch ];
```
Comment (PatrickDaG) [inline pkgs/by-name/ho/homer/package.nix:28]: From skimming the patch I'm not really sure what it does, or why its needed. Maybe add a short comment explaining to make it easier for future maintainers.

**Review by PatrickDaG — CHANGES_REQUESTED** [review]
Always thought homer was already packaged. LGTM.

**Comment by PatrickDaG** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354293`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>homer</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>homer</li>
  </ul>
</details>

File: pkgs/by-name/ho/homer/package.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  nativeBuildInputs = with pkgs; [
+    nodejs
+    dart-sass
+    pnpm.configHook
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    export SASS_EMBEDDED_BIN_PATH="${pkgs.dart-sass}/bin/sass"
```
Comment (SuperSandro2000) [inline pkgs/by-name/ho/homer/package.nix:39]:
```suggestion
    export SASS_EMBEDDED_BIN_PATH="${dart-sass}/bin/sass"
```

File: pkgs/by-name/ho/homer/package.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/by-name/ho/homer/package.nix:28]: It fixes the lock file probably because it was made with an older version of pnpm

Comment (Stunkymonkey) [reply pkgs/by-name/ho/homer/package.nix:30]: would this be enough: "# Enables specifying a custom Sass compiler binary path via `SASS_EMBEDDED_BIN_PATH` environment variable."

**Review by PatrickDaG — APPROVED** [review]

**Comment by PatrickDaG** [comment]
LGTM.

**Comment by Stunkymonkey** [comment]
@PatrickDaG i just updated the package again to its latest version.

**Comment by PatrickDaG** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354293`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>homer</li>
  </ul>
</details>

**Review by PatrickDaG — APPROVED** [review]

**Pull request merged**